src/libcollections/vec_deque.rs

```diff
@@ -1529,6 +1529,11 @@ pub struct IterMut<'a, T:'a> {
 }
 
 #[stable(feature = "rust1", since = "1.0.0")]
+unsafe impl<'a, T: Send+'a> Send for IterMut<'a, T> { }
+#[stable(feature = "rust1", since = "1.0.0")]
+unsafe impl<'a, T: Sync+'a> Sync for IterMut<'a, T> { }
```

Comment:
Done and then I tweaked the implementation to match `Iter`, which kills pointer all together.